vault/identity_store_entities.go

```diff
@@ -756,6 +780,33 @@ func (i *IdentityStore) mergeEntity(ctx context.Context, txn *memdb.Txn, toEntit
 			return errors.New("entity id to merge from does not belong to this namespace"), nil
 		}
 
+		// If we're not resolving a conflict, we check to see if
+		// any aliases conflict between the toEntity and this fromEntity:
+		if !forceMergeAliases && len(conflictingAliasIDsToKeep) == 0 {
+			for _, toAlias := range toEntity.Aliases {
+				for _, fromAlias := range fromEntity.Aliases {
+					// First, check to see if this alias clashes with an alias from any of the other fromEntities:
+					id, mountAccessorInAnotherFromEntity := fromEntityAccessors[fromAlias.MountAccessor]
```

Comment:
Where is fromEntityAccessors getting populated?

Reply:
A few lines below it, the full code block is this:
```
					id, mountAccessorInAnotherFromEntity := fromEntityAccessors[fromAlias.MountAccessor]
					if mountAccessorInAnotherFromEntity && (id != fromEntityID) {
						return fmt.Errorf("mount accessor %s found in multiple fromEntities, merge should be done with one fromEntity at a time", fromAlias.MountAccessor), nil
					}

					fromEntityAccessors[fromAlias.MountAccessor] = fromEntityID
```

Essentially, this block of code is to prevent the case where B and C are being merged into A, but B and C both have a clashing mountAccessor (even if A does not have that mount accessor).